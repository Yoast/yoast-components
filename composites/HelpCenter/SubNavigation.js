import React from "react";
import styled from "styled-components";
import colors from "../../style-guide/colors.json";

/**
 * Defines the styling of the SubNavigationElement.
 */
const SubNavigationElement = styled.nav`
	border-bottom: 1px solid ${ colors.$color_grey_light };
	font-size: 1em;
	text-transform: uppercase;
	display: block;
	
	ul {
		display: flex;
		flex-wrap: wrap;
		justify-content: center;
		list-style: none;
	    height: 100%;
	    padding: 0;
	    margin: 0;
	}
	
	li {
		display: inline-block;
	    text-align: center;
	    margin: 0 30px;
		
		&.tab-active {
	        box-shadow: inset 0 -5px 0 0 ${colors.$color_pink_dark}
		}
	}
	
	a {
		padding: 10px;
		display: flex;
		align-items: flex-start;
	}
`;

/**
 * Defines the style of the Link component.
 */
const Link = styled.a`
	color: ${colors.$color_pink_dark};
	text-decoration: none;
	display: block;
	padding: 18px 0;
`;

/**
 * Creates a SubNavigation component to be used for the navigation between tabs.
 */
export default class SubNavigation extends React.Component {

	/**
	 * Construct the component.
	 *
	 * @param {Object} props The props to use within this component.
	 *
	 * @constructor
	 */
	constructor( props ) {
		super( props );

		this.items = props.items;
	}

	/**
	 * Creates an LI element for every menu item passed to the component.
	 *
	 * @returns {Array} Array containing the rendered LI's.
	 */
	renderNavigationItems() {
		return this.items.map( ( item ) => {
			return (
				<li key={item.id} className={ item.url === this.props.activeTab ? "tab-active" : "" }>
					<Link href={item.url} onClick={ () => this.props.handler( item.url ) } > {item.label}</Link>
				</li>
			);
		} );
	}

	/**
	 * Renders the SubNavigation component.
	 *
	 * @returns {ReactElement} The rendered component.
	 */
	render() {
		return ( <SubNavigationElement>
			<ul>
				{ this.renderNavigationItems() }
			</ul>
		</SubNavigationElement> );
	}
}
