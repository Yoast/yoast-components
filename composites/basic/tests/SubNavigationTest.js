import React from "react";
import renderer from "react-test-renderer";

import SubNavigation from "../SubNavigation.js";

let items = [
	{ id: "tab1", label: "tab1", url: "http://localhost:3333/tab1", view: <p>This is some content for tab 1</p> },
	{ id: "tab2", label: "tab2", url: "http://localhost:3333/tab2", view: <p>This is some content for tab 2</p> },
	{ id: "tab3", label: "tab3", url: "http://localhost:3333/tab3", view: <p>This is some content for tab 3</p> },
];

test( "the SubNavigation matches the snapshot", () => {
	const component = renderer.create(
		<SubNavigation items={ items } activeTab="http://localhost:3333/tab1"/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
