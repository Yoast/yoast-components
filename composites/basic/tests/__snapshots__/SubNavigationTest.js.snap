// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the SubNavigation matches the snapshot 1`] = `
<nav
  className="sc-bdVaJa isMQIV"
>
  <ul>
    <li
      className="tab-active"
    >
      <a
        className="sc-bwzfXH hqQtJX"
        href="http://localhost:3333/tab1"
        onClick={[Function]}
      >
         
        tab1
      </a>
    </li>
    <li
      className=""
    >
      <a
        className="sc-bwzfXH hqQtJX"
        href="http://localhost:3333/tab2"
        onClick={[Function]}
      >
         
        tab2
      </a>
    </li>
    <li
      className=""
    >
      <a
        className="sc-bwzfXH hqQtJX"
        href="http://localhost:3333/tab3"
        onClick={[Function]}
      >
         
        tab3
      </a>
    </li>
  </ul>
</nav>
`;
